CMakeLists.txt

```diff
@@ -215,6 +217,10 @@ if(CMAKE_COMPILER_IS_CLANG)
   set(CLANG_LIBRARIES "stdc++")
 endif()
 
+if(CMAKE_COMPILER_IS_MINGW)
+  add_definitions(-DPCL_ONLY_CORE_POINT_TYPES)
+endif()
```

Comment:
This is hacky and silent :/

Reply:
I just followed this pattern. What would you recommend instead?

https://github.com/PointCloudLibrary/pcl/blob/e5b8fe5051283263097f843f635c7ee94a09483b/CMakeLists.txt#L146-L147

Reply:
In my ideal scenario we would not add it. `PCL_ONLY_CORE_POINT_TYPE` is an optional flag for users who want a smaller pre compilation footprint. We also use it for our CI for reduced testing builds, if I'm not mistaken.

But I wasn't aware this was being added by default on Windows and I'm concerned in case it actually needs to be. By design, it should not be a factor preventing compilation, hence my comment. It's a workaround, not really a fix.

I'm not deep into the problem be able to recommend any solution at this point.

Reply:
We enabled this for Windows long time ago because otherwise resulting DLLs contained too many symbols (or something like that).

Reply:
I agree that it's not the best way to fix it, but I think it's the only way short of separating code into multiple files, since the `/bigobj` MSVC flag and `-Wa,-mbig-obj` GCC flag doesn't seem to work.

See #897 for some more info regarding the issue. I'm not too sure what the issue is myself, something to do with the number of exported symbols or something akin to that.

Reply:
Hey guys, I ran into this issue before as well. In Windows,     _USE_MATH_DEFINES defines standard POSIX (*nix) constants like M_PI; you need it. /bigobj enables debug ELF output > than some size (forgot limit) - you need it to debug Windows/MSVC native apps. Curious why Mingw needs it however if it uses MSVC and MS CRT0 it would be the same as an x64-windows requirement

Reply:
FYI, Im out range testing 1.09/1.10 win64 right now! I can review the cmake changes next week when i get back to make sure no impact ...